### PR TITLE
EREGCSC-2352 -- Subjects page categories filter

### PR DESCRIPTION
**Pull request by PhilR8**
Resolves [EREGCSC-2352](https://jiraent.cms.gov/browse/EREGCSC-2352)

**Description**

When a researcher is looking at a long list of items on subject pages (or on future combined search pages), they may want to filter it down to a smaller set by choosing a specific document category.

**This pull request changes:**

- extends `content-search` endpoint to accept `categories` and `intcategories` parameters
   - the response contains all documents contained within the selected category AND that category's subcategories
- creates `fetch` Vue composable to encapsulate all front end fetching state management into a single location
- creates `removeList` Vue composable to allow each Vue view (Search, Subjects, Citations pages) to pass in a list of query parameters to remove from the `route` and the API call when filter criteria changes
   - Ex: each child component that updates the Vue Router adds and removes query parameters when filter criteria change.  
   - When selecting a Subject from the left sidebar Subject Picker, we want to remove `page` from the `route` -- we're getting a new list of results filtered by Subject and we want to send the user back to the first page of results.
   - This functionality -- scrubbing certain query parameters from the `route` and the API call based on filter changes -- was being individually specified in each filter component.  Now, each Vue view can determine its own behavior, create its own lists based on the AC for each filter, and then pass these lists to its child filter components.
- Adds API methods to fetch internal and external category trees.
   - Some minor refactoring and variable/method name changes to reflect that we are now fetching both Internal and External categories and category trees.
- Creates a Container Component to fetch these categories, transform them into their desired shape, and pass them down to a child UI component via slots
- Creates a Category filter dropdown/select Vue component that consumes the categories passed in from the Container Component
- CSS styling, minor refactoring around app where needed to accommodate new/updated methods described above
- Upgrades Cypress to to >= 13.7 (which in our case is 13.10) to fix bug that was causing some issues. See: https://docs.cypress.io/guides/references/changelog#13-7-0

**Steps to manually verify this change...**

1. Green check marks
2. Visit [Subjects page on the experimental deployment ](https://7harxb9cd6.execute-api.us-east-1.amazonaws.com/dev1312/subjects)
3. Note that you cannot see the Category Select on initial page load -- you must select a subject or enter a search query
   - Enter a search query and note that the Category Select appear
   - Select a category to refine your text search
   - Enter a different search query and note that the category is removed and the Select is reset
   - Further refine your search by selecting a subject
   - Select a category to refine your search within this selected subject
   - Choose a different subject and note that the category is removed and the Select is reset
4. Visit the subject page with a category already in the URL ([example](https://7harxb9cd6.execute-api.us-east-1.amazonaws.com/dev1312/subjects?q=timely&categories=11)) and note that the Category Select is pre-populated with the name of the category in the URL (Implementation Resources)
5. Ensure that documents that are not part of a subcategory are returned from the API along with documents that are part of a subcategory
6. Generally poke around, try the back and forward buttons, and make sure things work as outlined in the JIRA story and nothing breaks

### CODE REVIEW

File: solution/ui/e2e/cypress/support/e2e.js [hunk 1]
```diff
@@ -76,7 +73,12 @@ Cypress.Commands.add(
         cy.injectAxe();
         cy.checkA11y(
             subject,
-            { includedImpacts: ["critical", "serious"] },
+            {
+                includedImpacts: ["critical", "serious"],
+                rules: {
+                    "aria-progressbar-name": { enabled: false },
```
Comment (PhilR8) [inline solution/ui/e2e/cypress/support/e2e.js:79]: One of many Vuetify 3 things that are not yet A11y compliant

File: solution/ui/regulations/eregs-vite/src/components/dropdowns/Categories.vue [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    if (isIntcategories) {
+        // use regex to find a number of any length in a string that has the following pattern:
+        // subjects?subjects=2&intcategories=3
+        const intcategories = currentPopState.match(
+            /(?<=intcategories=)\d+/
+        )[0];
+        selectedId.value = `${intcategories}-intcategories`;
+    } else if (isCategories) {
+        // use regex to find a number of any length in a string that has the following pattern:
+        // subjects?subjects=2&categories=3
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/dropdowns/Categories.vue:143]: these comments were Copilot prompts and are retained

File: solution/ui/e2e/package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
   "devDependencies": {
     "@testing-library/cypress": "^10.0.1",
     "axe-core": "^4.6.3",
-    "cypress": "^13.2.0",
+    "cypress": "^13.10.0",
```
Comment (PhilR8) [inline solution/ui/e2e/package.json:15]: Upgraded to >= 13.7 to fix bug that was causing some issues:

> Changed screenshot capture behavior in Chromium to activate the main Cypress tab before capturing. This prevents screenshot capture from timing out in certain situations. Fixed in [#29038](https://github.com/cypress-io/cypress/pull/29038). Fixes [#5016](https://github.com/cypress-io/cypress/issues/5016)

File: solution/ui/regulations/eregs-vite/src/components/subjects/PolicySelections.vue [hunk 4]
```diff
@@ -10,30 +12,22 @@ const $route = useRoute();
...
-            ? { [type]: filteredParamsArray }
-            : {};
+const removeClick = () => {
+    const routeClone = { ...$route.query };
 
-    delete routeClone[type];
+    const cleanedRoute = useRemoveList({
+        route: routeClone,
+        removeList,
+    });
```
Comment (PhilR8) [inline solution/ui/regulations/eregs-vite/src/components/subjects/PolicySelections.vue:25]: cleaned up a bunch of superfluous logic here

**Review by peggles2 — APPROVED** [review]
LGTM! 👍

**Pull request merged**